File: Math/547-findEvenNumber.py
from collections import Counter
class Solution:
    def findEvenNumbers(self, digits):
        c=Counter(digits)
        ans=[]
        for n in range(100,999,2):
            if n&1==0:
                cur=[int(i) for i in str(n)]
                isAllInDigits=True;
                tStr=Counter(cur)
                for k,v in tStr.items():
                    if v>c[k]:
                        isAllInDigits=False
                        continue
                if isAllInDigits:
                    ans.append(n)
        return ans
    def findEvenNumbers2(self, digits):
        C=Counter(digits)
        ans=[]
        for n in range(100,999,2):
            a=n//100
            b=(n//10)%10
            c=n%10
            if Counter((a,b,c))<=C:
                ans.append(n)
        return ans
nums=[2,1,3,0]
print(Solution().findEvenNumbers(nums))

File: Math/547-findEvenNumber.cpp
#include<iostream>
#include<vector>
#include<map>

using std::cout;
using VEC=std::vector<int>;
using MAP=std::map<int,int>;

MAP myCounter(const VEC& nums)
{
    MAP ans{};
    for(auto &i:nums)
        ans[i]+=1;
    return ans;
}

VEC findEvenNumbers(const VEC& digits)
{
    VEC ans{};
    MAP total{myCounter(digits)};

    for(int i=100;i<999;i+=2)
    {
        VEC tmp{i/100,(i/10)%10,i%10};
        MAP m{myCounter(tmp)};
        bool isCovered=true;
        for(auto &[k,v]:m)
        {
            if(v>total[k])
                isCovered=false;
        }
        if(isCovered)
            ans.push_back(i);
    }
    return ans;
}

int main()
{
    VEC nums{2,1,3,0};
    VEC res=findEvenNumbers(nums);
    for(auto &v:res)
        cout<<v<<',';
}